### ISSUE TITLE
Assertion `functionList && "not in a function"' failed.

### ISSUE THREAD

**Issue by sscalpone** [flang-compiler/f18-llvm-project#508]
```
% cat cdir.f90
!CRAY - The following compiler directive assures that the multiply
!       in the internal function is done with 64 bits on the CRAY
!       See below for more details
!DIR$ INTEGER=64
```

```
% bbc cdir.f90 |& head
bbc: /local/home/sjs/mirror/fir-dev-mirror/flang/lib/Lower/PFTBuilder.cpp:284: Fortran::lower::pft::Evaluation& {anonymous}::PFTBuilder::addEvaluation(Fortran::lower::pft::Evaluation&&): Assertion `functionList && "not in a function"' failed.
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
Stack dump:
0.      Program arguments: bbc cdir.f90
 #0 0x0000557b20e1c8fd llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/local/home/sjs/mirror/install/bin/bbc+0x345e8fd)
 #1 0x0000557b20e1a634 llvm::sys::RunSignalHandlers() (/local/home/sjs/mirror/install/bin/bbc+0x345c634)
 #2 0x0000557b20e1a7a0 SignalHandler(int) (/local/home/sjs/mirror/install/bin/bbc+0x345c7a0)
 #3 0x00007f4e956418a0 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x128a0)
 #4 0x00007f4e942adf47 raise /build/glibc-2ORdQG/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x00007f4e942af8b1 abort /build/glibc-2ORdQG/glibc-2.27/stdlib/abort.c:81:0
```

**Comment by jeanPerier** [comment]
PFT bug, we need to support top level CompilerDirective.

The PFT analysis does not expect that CompilerDirective can be ProgramUnit, it expects that they are nested inside a ProgramUnit. 

`f18 -fdebug-dump-parse-tree` on the bug program gives:
```
Program -> ProgramUnit -> CompilerDirective -> NameValue
| Name = 'integer'
| uint64_t = '64'
```

The fact that CompilerDirective can be ProgramUnit is new from 2 month ago: This is new from https://reviews.llvm.org/D86555
The PFT was not aligned to support this.
We need to update the PFT to capture and (for now) ignore this.

**Comment by jeanPerier** [comment]
Fixed by #525
